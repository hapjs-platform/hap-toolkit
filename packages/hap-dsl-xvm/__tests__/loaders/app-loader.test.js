/*
 * Copyright (c) 2021-present, the hapjs-platform Project Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

const path = require('path')
const { rowify } = require('hap-dev-utils')
const appLoader = require('../../lib/loaders/app-loader').default

describe('app-loader', () => {
  const mockLoader = {
    context: __dirname,
    rootContext: __dirname,
    resourceQuery: `?uxType=app`,
    resourcePath: path.resolve(__dirname, 'app.ux')
  }
  const samples = [
    [
      'normal',
      `
    <script>
    export default {
      method1() {
        console.log('this is method1')
      }
    }
    </script>
      `
    ],
    [
      'without <script/> tag',
      `
      <!-- no script tag -->
      `
    ]
  ]
  beforeEach(() => {
    Object.assign(mockLoader, {
      emitError: jest.fn(),
      emitWarning: jest.fn()
    })
  })

  it.each(samples)('%s', async (_, source) => {
    const loader = Object.assign({}, mockLoader)
    const result = await appLoader.call(loader, source)

    expect(rowify([source, result])).toMatchSnapshot('source-vs-result')
    expect(loader.emitError.mock.calls).toMatchSnapshot('emitted errors')
    expect(loader.emitWarning.mock.calls).toMatchSnapshot('emitted warnings')
  })
})
