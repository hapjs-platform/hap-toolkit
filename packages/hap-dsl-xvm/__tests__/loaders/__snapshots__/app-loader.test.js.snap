// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-loader normal: emitted errors 1`] = `Array []`;

exports[`app-loader normal: emitted warnings 1`] = `Array []`;

exports[`app-loader normal: source-vs-result 1`] = `
"
    <script>
    export default {
      method1() {
        console.log('this is method1')
      }
    }
    </script>
      
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

var $app_style$ = {}
var $app_script$ = require(\\"!../../lib/loaders/script-loader.js!../../../../node_modules/@hap-toolkit/packager/lib/loaders/module-loader.js!../../../../node_modules/@hap-toolkit/packager/lib/loaders/manifest-loader.js?path=<CWD>/packages/hap-dsl-xvm/__tests__/loaders!../../../../node_modules/babel-loader/lib/index.js?cwd=<CWD>&cacheDirectory&comments=false&configFile=<CWD>/babel.config.js!../../lib/loaders/fragment-loader.js?index=0&type=script!./app.ux?uxType=app\\")

$app_define$('@app-application/app', [], function ($app_require$, $app_exports$, $app_module$) {
  
  $app_script$($app_module$, $app_exports$, $app_require$)
  if ($app_exports$.__esModule && $app_exports$.default) {
    $app_module$.exports = $app_exports$.default
  }
  $app_module$.exports.manifest = require(\\"./manifest.json\\")
  $app_module$.exports.style = { list: [ $app_style$ ] }
  
})
$app_bootstrap$('@app-application/app', { packagerVersion: QUICKAPP_TOOLKIT_VERSION })
"
`;

exports[`app-loader without <script/> tag: emitted errors 1`] = `Array []`;

exports[`app-loader without <script/> tag: emitted warnings 1`] = `Array []`;

exports[`app-loader without <script/> tag: source-vs-result 1`] = `
"
      <!-- no script tag -->
      
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

var $app_style$ = {}
var $app_script$ = null
$app_define$('@app-application/app', [], function ($app_require$, $app_exports$, $app_module$) {
  
})
$app_bootstrap$('@app-application/app', { packagerVersion: QUICKAPP_TOOLKIT_VERSION })
"
`;
